velox/functions/lib/benchmarks/LikeFunctionsBenchmark.cpp

```diff
@@ -0,0 +1,218 @@
+DEFINE_int32(vector_size, 100, "Vector size");
+DEFINE_int32(num_rows, 10000, "Number of rows");
+
+namespace facebook::velox::functions::test {
+
+class LikeFunctionsBenchmark : public functions::test::FunctionBaseTest {
+ public:
+  explicit LikeFunctionsBenchmark() {
+    exec::registerStatefulVectorFunction("like", likeSignatures(), makeLike);
+  }
+
+  std::string generateRandomString(
+      const char* charArr,
+      std::optional<vector_size_t> length = 10) {
+    vector_size_t arrLen = strlen(charArr);
+    vector_size_t len;
+    if (length) {
+      len = *length;
+    } else {
+      vector_size_t minLen = 1;
+      vector_size_t maxLen = 10;
+      len = rand() % maxLen + minLen;
+    }
+    std::string output;
+
+    for (int i = 0; i < len; i++) {
+      output += charArr[rand() % arrLen];
+    }
+    return output;
+  }
+
+  std::string generatePattern(
+      PatternKind patternKind,
+      std::string inputString) {
+    switch (patternKind) {
+      case PatternKind::kExactlyN:
+        return generateRandomString(kSingleWildcardCharSet, inputString.size());
+      case PatternKind::kAtLeastN:
+        return generateRandomString(kWildcardCharSet);
+      case PatternKind::kPrefix: {
+        auto fixedPatternLength =
+            std::min(vector_size_t(inputString.size()), 10);
+        auto fixedPatternString = inputString.substr(0, fixedPatternLength);
+        return fixedPatternString + generateRandomString(kAnyWildcardCharSet);
+      }
+      case PatternKind::kSuffix: {
+        auto fixedPatternStartIdx =
+            std::max(vector_size_t(inputString.size() - 10), 0);
+        auto fixedPatternString = inputString.substr(fixedPatternStartIdx, 10);
+        return generateRandomString(kAnyWildcardCharSet) + fixedPatternString;
+      }
+      default:
+        return inputString;
+    }
+  }
+
+  void run(FlatVector<StringView>* input, StringView patternString) {
+    auto like = [&](std::optional<std::string> str,
+                    std::optional<std::string> pattern) {
+      return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+    };
+
+    folly::BenchmarkSuspender kSuspender;
+    FunctionBenchmarkBase benchmarkBase;
+    kSuspender.dismiss();
+    vector_size_t vectorSize = input->size();
+    for (int i = 0; i < FLAGS_num_runs; i++) {
+      for (vector_size_t j = 0; j < vectorSize; j++) {
+        auto inputAtIdx = input->valueAt(j);
+        like(inputAtIdx, patternString);
+      }
+    }
+  }
+
+  void run(PatternKind patternKind) {
+    auto like = [&](std::optional<std::string> str,
+                    std::optional<std::string> pattern) {
+      return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+    };
+
+    folly::BenchmarkSuspender kSuspender;
+    FunctionBenchmarkBase benchmarkBase;
+    kSuspender.dismiss();
+    VectorFuzzer::Options opts;
+    opts.vectorSize = FLAGS_vector_size;
+    VectorFuzzer fuzzer(opts, benchmarkBase.pool());
```

Comment:
Avoid constructing the fuzzer for each itteration. maybe move them to the clas.

Reply:
Thanks for the suggestion, moved fuzzer generation part to the constructor